### DISCUSSION TITLE
Problems with Deployment of Kafka-Openstack

### DISCUSSION THREAD

**Discussion by Bahncard** [eclipse-xpanse/xpanse#1367] [Q&A] [answered]
Hi team,

I hope you're all having a great day!

I'm facing some issues while trying to deploy Kafka on OpenStack using Xpanse. Here are the details:

## Checklist

 - [x] I could not find a solution in the documentation, the issues or discussions

## Problem Description

When attempting to add a credential with the user role, any combination of username, password, and domain name seems to successfully add a credential.

Upon adding a credential for OpenStack and updating the OS_AUTH_URL variable in the openstack-Kafka.yml file, the deployment of Kafka on OpenStack fails with the following error message:
Deployment Failed.
``` TFExecutor Exception:TFExecutor.tfPlan failed. Error: Error creating OpenStack compute client: Authentication failed with data.openstack_compute_availability_zones_v2.osc-az, on [resources.tf](http://resources.tf/) line 112, in data "openstack_compute_availability_zones_v2" "osc-az": 112: data "openstack_compute_availability_zones_v2" "osc-az" {} Error: Error creating OpenStack image client: Authentication failed with data.openstack_images_image_v2.image, on [resources.tf](http://resources.tf/) line 132, in data "openstack_images_image_v2" "image": 132: data "openstack_images_image_v2" "image" { ```

## Screenshots
![deploy-failed](https://github.com/eclipse-xpanse/xpanse/assets/87471456/4dfbbcff-daaf-4e92-921a-2bf4b90f7c63)
![kafka-yml](https://github.com/eclipse-xpanse/xpanse/assets/87471456/1e830948-e306-4d00-b763-80d41133c3fb)
![rc-file](https://github.com/eclipse-xpanse/xpanse/assets/87471456/007d1928-0a86-4736-86a7-8f2ed8c41a16)

**Comment by swaroopar** [comment, marked as answer]
@Bahncard 
We have fixed this issue with PR https://github.com/eclipse-xpanse/xpanse/pull/1372 . 
Now the Openstack plugin can handle user and project connected to different domains
